code/modules/fabrication/fabricator_topic.dm

```diff
@@ -39,6 +39,17 @@
 		if(CanInteract(user, DefaultTopicState()))
 			filter_string = new_filter_string
 			. = TOPIC_REFRESH
+	
+	//Tab expanding/collapsing
+	if(href_list["toggle_resources"])
+		ui_expand_resources = !ui_expand_resources
+		. = TOPIC_REFRESH
+	if(href_list["toggle_queue"])
+		ui_expand_queue = !ui_expand_queue
+		. = TOPIC_REFRESH
+	if(href_list["toggle_config"])
+		ui_expand_config = !ui_expand_config
+		. = TOPIC_REFRESH
```

Comment:
For what it's worth, this kind of global state is problematic. I know other nanoui stuff does this (I probably wrote some) so I won't be too critical.

Reply:
Yeah, not a fan either. Its just we got limited tools to work with that. Would need cookies or something to store the prefs client side. And my knowledge of JS is kinda limited XD